cijoe/scripts/latency_plotter.py

```diff
@@ -38,143 +38,179 @@ def data_as_a_function_of(data, x="iodepth", y="lat", filter=lambda _: True) ->
+            framealpha=1,
+        )
+
+    if xlabel:
+        plt.xlabel(xlabel)
 
-    for ident, samples in data.items():
-        grouped[samples["group"]][ident] = samples
-    return grouped
+    plt.xticks(
+        list(x_ticks.values()),
+        list(x_ticks.keys()),
+        rotation=50 if is_grouped else 0,
+        ha="right" if is_grouped else "center",
+    )
 
+    plt.ylabel("nanoseconds")
+    plt.ylim([0, y_limit])
 
-def line_plot(args, cijoe, step):
+
+def create_plots(args, cijoe, step):
     search = step.get("with", {}).get("path", args.output)
     if not search:
         return errno.EINVAL
+    artifacts = args.output / "artifacts"
 
     search_for = FIO_OUTPUT_NORMALIZED_FILENAME
 
     path = next(Path(search).rglob(search_for))
-
     with path.open() as jfd:
         data = json.load(jfd)
 
-    plot_attributes = plot_attributes_from_step(step)
+    plot_attributes = get_plot_attributes(step)
 
-    x = "iodepth"
     y = "lat"
 
-    data = data_as_a_function_of(data, x, y)
-
-    grouped = build_plot_groups(data)
-
-    for group, dset in grouped.items():
-        plt.clf()
-
-        for _, samples in dset.items():
-            label = samples["label"]
-            attrs = plot_attributes["legends"].get(label, None)
-            if attrs is None:
-                log.error(f"Missing plot-attributes for label({label})")
-                continue
-
-            xs, ys = [list(t) for t in zip(*sorted(zip(samples["xs"], samples["ys"])))]
-            plt.plot(
-                xs,
-                ys,
-                label=attrs["legend"],
-                color=attrs["color"],
-                marker=attrs["marker"],
-            )
-
-        plt.xlabel(x)
-        plt.ylabel(y)
-
-        plt.legend(
-            bbox_to_anchor=(0, 0.95, 1, 0.2),
-            loc="lower center",
-            ncol=2,
-            facecolor="white",
-            framealpha=1,
+    max_y = max(map(lambda item: item[1][y], data))
+    all_groups = set(map(lambda item: item[1]["ctx"]["group"], data))
+
+    # Create a plot for each group, showing the scalability of the xNVME overhead
+    for group in all_groups:
+        dset = data_as_a_function_of(
+            data,
+            x="iodepth",
+            y=y,
+            filter=lambda item: item["ctx"]["group"] == group
+            and item["ctx"]["iosize"] == 4096,
+        )
+        draw_bar_plot(dset, plot_attributes, xlabel="iodepth", y_limit=max_y * 1.3)
+
+        os.makedirs(artifacts, exist_ok=True)
+        plt.savefig(artifacts / f"fio_barplot_{group}_iodepth.png")
+
+    for group in all_groups:
+        dset = data_as_a_function_of(
+            data,
+            x="iosize",
+            y=y,
+            filter=lambda item: item["ctx"]["group"] == group
+            and item["ctx"]["iodepth"] == 1,
         )
-        plt.ylim(plot_attributes["limits"]["lineplot"]["y_lim"])
+        draw_bar_plot(dset, plot_attributes, xlabel="iosize", y_limit=max_y * 1.3)
```

Comment:
Maybe this scale constant (`1.3`) which is used many places should be defined as a constant?